templates/common/_base/units/machine-config-daemon-firstboot.service.yaml

```diff
@@ -16,6 +16,9 @@ contents: |
   # Disable existing repos (if any) so that OS extensions would use embedded RPMs only
   ExecStartPre=-/usr/bin/sh -c "sed -i 's/enabled=1/enabled=0/' /etc/yum.repos.d/*.repo"
   ExecStart=/run/bin/machine-config-daemon firstboot-complete-machineconfig
+  {{if isTechPreviewNoUpgradeEnabled . -}}
+  ExecStartPost=/usr/bin/mkdir -p /run/cluster-api && /usr/bin/touch /run/cluster-api/bootstrap-success.complete
+  {{end -}}
```

Comment:
Can't this be in Go as part of the `firstboot-complete-machineconfig` process?  We have too much load-bearing bash script run as root today.

Also, regardless of that change, can you please add a comment in the code with a link to whatever process/code is reading this file?  Is it something in kubelet?

Reply:
Also, is it actually necessary to make this conditional on the tech preview flag?  Does anything read it if the node wasn't created via cluster-api?